truelearn/models/_knowledge.py

```diff
@@ -89,63 +86,143 @@ def mean(self) -> float:
-    update(topic_id, kc)
-        Update the AbstractKnowledgeComponent associated with the topic_id
-    topic_kc_pairs()
-        Return an iterable of (topic_id, AbstractKnowledgeComponent) pairs
-    knowledge_components()
-        Return an iterable of AbstractKnowledgeComponents.
+    In TrueLearn, we assume every learner's knowledge consists of many different
+    knowledge components. The Knowledge class is used to represent this relationship.
 
+    The class can be used to represent 1) the learner's knowledge and
+    2) the knowledge of a learnable unit.
     """
 
-    def __init__(self, knowledge: dict[Hashable, AbstractKnowledgeComponent] | None = None) -> None:
+    def __init__(
+        self,
+        knowledge: Optional[Dict[Hashable, AbstractKnowledgeComponent]] = None,
+    ) -> None:
+        """Init the Knowledge object.
+
+        If the given knowledge is None, the knowledge will be initialized emptily.
+
+        Args:
+            knowledge: A dict mapping a hashable id of the knowledge component
+                to the corresponding knowledge component object. Defaults to None.
+        """
         super().__init__()
 
-        if knowledge is not None:
-            self.__knowledge = knowledge
-        else:
-            self.__knowledge: dict[Hashable, AbstractKnowledgeComponent] = {}
+        if knowledge is None:
+            knowledge = {}
+        self.__knowledge = knowledge
+
+    def get_kc(
+        self, topic_id: Hashable, default: AbstractKnowledgeComponent
+    ) -> AbstractKnowledgeComponent:
+        """Get the knowledge component associated with the given id.
 
-    def get_kc(self, topic_id: Hashable, default: AbstractKnowledgeComponent) -> AbstractKnowledgeComponent:
+        Args:
+            topic_id: The id that uniquely identifies a knowledge component.
+            default: The default knowledge component to return.
+
+        Returns:
+            The knowledge component extracted from the knowledge
+            if the topic_id exists in the knowledge.
+            Otherwise, the default value will be returned.
+        """
         return self.__knowledge.get(topic_id, default)
 
     def update_kc(self, topic_id: Hashable, kc: AbstractKnowledgeComponent) -> None:
+        """Update the knowledge component associated with the given topic_id.
+
+        If the topic_id doesn't exist in the AbstractKnowledge, the mapping
+        from the topic_id to the knowledge component will be created.
+
+        Args:
+          topic_id: Hashable:
+          kc: AbstractKnowledgeComponent:
+        """
         self.__knowledge[topic_id] = kc
 
-    def topic_kc_pairs(self) -> Iterable[tuple[Hashable, AbstractKnowledgeComponent]]:
+    def topic_kc_pairs(
+        self,
+    ) -> Iterable[Tuple[Hashable, AbstractKnowledgeComponent]]:
+        """Return an iterable of the (topic_id, knowledge_component) pair."""
         return self.__knowledge.items()
```

Comment:
I suspect this is the pycharm bug eventhough I updated of:
Expected type 'Iterable[tuple[Hashable, AbstractKnowledgeComponent]]', got '_dict_items[Hashable, AbstractKnowledgeComponent]' instead

Reply:
Yeah. It seems that this doesn't cause any problem in my machine and Github Action.